cmd/azure-config-credentials-injector/main.go

```diff
@@ -71,30 +73,47 @@ func mergeCloudConfig(_ *cobra.Command, args []string) error {
 		return err
 	}
 
-	azureClientId, found := mustLookupEnvValue(clientIDEnvKey)
-	if !found {
-		return fmt.Errorf("%s env variable should be set up", clientIDEnvKey)
+	// Read credentials from mounted Secret files
+	azureClientId, err = readSecretFile("/etc/azure/credentials/azure_client_id")
```

Comment:
Should this be more like
```suggestion
	azureClientId, err = readSecretFile(fmt.Sprintf("%s/azure_client_id", credentialsPath)
```